compiler/modules/front/expr.bal

```diff
@@ -836,7 +836,13 @@ function codeGenBitwiseBinaryExpr(ExprContext cx, bir:BasicBlock bb, s:BinaryBit
     ValueFlags resultFlags = leftFlags & rightFlags;
     t:SemType lt = t:widenUnsigned(lhs.semType);
     t:SemType rt = t:widenUnsigned(rhs.semType);
-    t:SemType resultType = op == "&" ? t:intersect(lt, rt) : t:union(lt, rt);
+    t:SemType resultType;
+    if op is "<<"|">>"|">>>" {
+        resultType = op is ">>"|">>>" && t:isSubtype(cx.mod.tc, lt, t:intWidthUnsigned(32)) ? lt : t:INT;
```

Comment:
Think about case where lt is a proper subtype of unisgnedN.

Reply:
This is correct since `lt` is the result of [`widenUnsigned`](https://github.com/ballerina-platform/nballerina/blob/2d43f999afe0c44e4be2b55b575b15537c8ac99b/compiler/modules/types/core.bal#L978) (add a test case to cover this)